### PR TITLE
Zi/make vendor right

### PR DESCRIPTION
**Pull request by lziest**
1. Remove root vendor and move it to cmd/, create a symbolic link for gokeyless itself under cmd/vendor/github.com/cloudflare/gokeyless
2. fix another data race so CI is green.

### CODE REVIEW

**Comment by lziest** [comment]
@FiloSottile please review.

File: Makefile [hunk 1]
```diff
@@ -33,8 +33,12 @@ install-config:
 	@install -m400 pkg/testing-ecdsa.key $(CONFIG_PREFIX)/keys/testing-ecdsa.key
 	@install -m400 pkg/testing-rsa.key $(CONFIG_PREFIX)/keys/testing-rsa.key
 
-$(INSTALL_BIN)/$(NAME): | install-config
-	@GOOS=linux GOARCH=$(ARCH) go build -o $@ ./cmd/$(NAME)/...
+.PHONY: get-gvt
+get-gvt:
+	go get -u github.com/FiloSottile/gvt
```
Comment (jkroll-cf) [inline Makefile]: Trying to promote a world where we git clone && git checkout -f <revision>, since `go get` doesn't have much of an integrity story.

Comment (FiloSottile) [inline Makefile]: I agree with the effort 100%, but without aggressive fsck settings, git offers no integrity against malicious adversaries.

If the goal is just to pin the version, I think that's not what that make target wants.


Sent from a small keyboard

**Comment by lziest** [comment]
this is kinda urgent

File: cmd/gokeyless-cli-tester/vendor/github.com/cloudflare/gokeyless [hunk 2]
```diff
@@ -0,0 +1 @@
+../../../../../../gokeyless
```
Comment (FiloSottile) [inline cmd/gokeyless-cli-tester/vendor/github.com/cloudflare/gokeyless]: This should be `../../../../..` in case the folder is not called gokeyless.

**Comment by FiloSottile** [comment]
I'm generally not a fan of `gvt restore` as it breaks `go get`, but your call. If you keep it, I'd change https://github.com/cloudflare/gokeyless#source-installation to mandate the Makefile though.

`fix another data race on hashmap` seems unrelated.

**Comment by FiloSottile** [comment]
To avoid triple bloat, you can put `vendor/` in `cmd/`.

**Comment by lziest** [comment]
I tried `vendor/` under `cmd/` but it doesn't work with `go build cmd/gokeyless` then, I haven't tried `go build cmd/...` though

**Comment by FiloSottile** [comment]
That's strange, it's really supposed to work the same however command you use to build.

Try just moving the vendor folder up one level and then running the exact same "go build".

**Comment by lziest** [comment]
It's probably I didn't do the path symlink right. Now it is working with cmd/vendor

**Comment by lziest** [comment]
And the error message was complaining about gokeyless/client.go can't find dependency packages

**Comment by lziest** [comment]
@kisom @FiloSottile @jkroll-cf review it again?

**Comment by jkroll-cf** [comment]
Not being super clear on vendor rules or best practices, if @FiloSottile is happy and the code builds, I'm happy. In the longer future, we should work to isolate the functionality we need to use as a library into packages that we can treat as libraries everywhere, which will moot the underlying problem.

LGTM, for the moment.

**Comment by lziest** [comment]
I confirm that when I vendor and import github.com/clouflare/gokeyless, the log package being picked up by gokeyless is now locally 'go get' or vendored cfssl/log, not the nested vendored one.

**Pull request merged**